libs/backend/src/exporter.ts

```diff
@@ -19,11 +20,10 @@ const MAXIMUM_FAT32_FILE_SIZE = 2 ** 32 - 1;
 /**
  * Possible export errors.
  */
-export type ExportDataError =
-  | { type: 'relative-file-path'; message: string }
-  | { type: 'permission-denied'; message: string }
-  | { type: 'file-system-error'; message: string }
-  | { type: 'missing-usb-drive'; message: string };
+export interface ExportDataError {
+  type: BaseExportDataError;
+  message: string;
```

Comment:
Left the type exported by the generic `exporter.ts` as is, with its `message` field intact, since I didn't want to balloon the scope of this PR